src/python/pants/backend/python/lint/python_fmt.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from dataclasses import dataclass
+from typing import Iterable, List, Type
+
+from pants.backend.python.rules.targets import PythonSources
+from pants.engine.fs import Digest, Snapshot
+from pants.engine.objects import union
+from pants.engine.rules import UnionMembership, UnionRule, rule
+from pants.engine.selectors import Get
+from pants.rules.core.determine_source_files import AllSourceFilesRequest, SourceFiles
+from pants.rules.core.fmt import (
+    FmtConfigurations,
+    FmtResult,
+    LanguageFmtResults,
+    LanguageFmtTargets,
+)
+
+
+@dataclass(frozen=True)
+class PythonFmtTargets(LanguageFmtTargets):
+    required_fields = (PythonSources,)
+
+
+@union
+class PythonFmtConfigurations:
+    pass
```

Comment:
I spent some time debating if this should subclass `FmtConfigurations` and the Python formatters should subclass this, like `class BlackConfigurations(PythonFmtConfigurations)`. 

I decided no because "flat is better than nested", i.e. I think this is a simpler implementation by reducing the size of the inheritance chain.